app/public/js/common/configLocation.js

```diff
@@ -38,12 +38,26 @@ const configuration = {
       userConfigPath = `${userHome}/Library/Preferences/Soundnode`;
     }
 
+    // Guard to assert type of string.
+    if (typeof userConfigPath !== "string") {
```

Comment:
can you tell me if there is a specific case for this check to be necessary? thanks

Reply:
Since `userConfigPath` is initially set to `null`, if the platform is not linux, windows, or darwin, then it never gets assigned to a `string`. Thus if `userConfigPath` is still a null value the platform is obviously not supported, so an exception is thrown to let the user know why the `userConfigPath` value could not be set properly.